### PR TITLE
RotateObject action

### PR DESCRIPTION
**Pull request by jaudick**
[UNITY](https://github.com/NextCenturyCorporation/ai2thor/pull/196)

I'm not sure if I handled an input bool parameter for an action properly since this appears to be the first action with a bool parameter. This was how I got it to work:  take in a string of: "True/False" or "true/false" and convert to a bool whenever its not a bool. 

I also added controller tests for TorqueObject. I never made any when I merged that branch and felt it was necessary since this branch originally branched from the TorqueObject branch. The PushObject controller test also did not have a test with a default force parameter so I added one test in its test_controller method since I was already touching the force code of TorqueObject here

### CODE REVIEW

File: tests/test_parameter.py [hunk 1]
```diff
@@ -236,6 +236,50 @@ def test_get_force(self):
...
+        self.assertIsInstance(clockwise, bool)
+        self.assertEqual(clockwise, False)
+
+        clockwise = self.parameter_converter._get_clockwise(clockwise=True)
+        self.assertIsInstance(clockwise, bool)
+        self.assertEqual(clockwise, True)
+
+        clockwise = self.parameter_converter._get_clockwise()
+        self.assertIsInstance(clockwise, bool)
+        self.assertEqual(clockwise, True)
```
Comment (ThomasSchellenbergNextCentury) [inline tests/test_parameter.py:266]: I know you were questioning your input handling, but the fact that you have these good unit tests makes me pretty confident you're handling it well :)

File: machine_common_sense/parameter.py [hunk 2]
```diff
@@ -251,6 +271,8 @@ def _validate_and_convert_params(self, **kwargs) -> Tuple:
         }
         move_magnitude = self._get_move_magnitude(action, force, amount)
         (teleport_rotation, teleport_position) = self._get_teleport(**kwargs)
+        clockwise = self._get_clockwise(**kwargs)
+        (teleport_rotation, teleport_position) = self._get_teleport(**kwargs)
```
Comment (ThomasSchellenbergNextCentury) [inline machine_common_sense/parameter.py:275]: I think you accidentally copied this one line (compare 273 and 275)

File: machine_common_sense/action.py [hunk 3]
```diff
@@ -464,6 +464,53 @@ class Action(Enum):
...
 
+    ROTATE_OBJECT = (
+        "RotateObject",
+        "9",
+        "Apply rotation to a nearby object. (objectId=string, " +
+        "clockwise=bool(default:True), objectImageCoordsX=float, " +
+        "objectImageCoordsY=float)"
+    )
+    """
+    Apply rotation to a nearby object.
```
Comment (ThomasSchellenbergNextCentury) [inline machine_common_sense/action.py:475]: Please change this to: `Apply a rotation of 5 degrees to a nearby object.` (so it includes the rotation amount)

File: machine_common_sense/parameter.py [hunk 4]
```diff
@@ -66,7 +68,7 @@ class Parameter:
     # # TODO: Move this to an enum or some place, so that you can determine
     # # special move interactions that way
     FORCE_ACTIONS = ["PushObject", "PullObject", "TorqueObject"]
-    OBJECT_MOVE_ACTIONS = ["CloseObject", "OpenObject"]
+    OBJECT_MOVE_ACTIONS = ["CloseObject", "OpenObject", "RotateObject"]
```
Comment (ThomasSchellenbergNextCentury) [inline machine_common_sense/parameter.py:71]: I don't think this needs to be here, does it? The rotation amount is always the same so it doesn't need those input checks.

Comment (jaudick) [reply machine_common_sense/parameter.py]: That makes sense, I made the requested changes

**Comment by kdrumm** [comment]
Please make sure the unit tests pass

**Review by deanwetherby — APPROVED** [review]
I like all the unit tests.

**Review by kdrumm — APPROVED** [review]

**Pull request merged**